lib/src/controllers/cartesian_controller.dart

```diff
@@ -1,12 +1,44 @@
+import 'dart:math';
+
 import 'package:chart_it/src/animations/chart_animations.dart';
+import 'package:chart_it/src/charts/data/bars/bar_series.dart';
 import 'package:chart_it/src/charts/data/core/cartesian/cartesian_data.dart';
+import 'package:chart_it/src/charts/painters/cartesian/bar_painter.dart';
+import 'package:chart_it/src/charts/painters/cartesian/cartesian_painter.dart';
+import 'package:chart_it/src/extensions/data_conversions.dart';
+import 'package:chart_it/src/extensions/data_mixins.dart';
+import 'package:chart_it/src/extensions/primitives.dart';
 import 'package:flutter/material.dart';
 
 class CartesianController extends ChangeNotifier
-    with ChartAnimationsMixin<CartesianSeries> {
+    with
+        CartesianDataMixin<CartesianSeries>,
+        ChartAnimationsMixin<CartesianSeries> {
+  final Map<CartesianSeries, CartesianConfig> _seriesConfigs = {};
+  final Map<Type, CartesianPainter> painters = {};
+
   List<CartesianSeries> currentData = List.empty();
   List<CartesianSeries> data;
```

Comment:
Seems like this data is not actually used anywhere. Check if we can get rid of it?

Reply:
Used in `CartesianChartPainter` so we require it